src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,542 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  This workload compares the query performance of SSB (star schema benchmark) queries with columnar indexes to performances without them.
+
+  The workload consists of the following phases and actors:
+    1. Run SSB queries with columnstore indexes.
+    2. Drop columnstore indexes.
+    3. Run SSB queries without columnstore indexes.
+
+  The DSI test_control will drop caches inbetween each test to measure cold cache performance.
+
+Keywords:
+- columnstore
+- analytics
+- SSB
+
+Clients:
+  Default:
+    QueryOptions:
+      # Allow for longer duration since column store index build may take a while.
+      socketTimeoutMS: 3_600_000  # = 1 hour
+      connectTimeoutMS: 3_600_000
+
+# Defines the database name.
+Database: &db ssb
+
+# Defines the collection name.
+Collection: &Collection order_lineitem
+
+Q1_1Query: &Q1_1Query
+  Pipeline:
+    [
+      {
+        "$match": {
+          "order.o_orderdate": {
+            "$gte": ISODate("1992-01-01T00:00:00Z"),
+            "$lte": ISODate("1996-01-01T00:00:00Z")
+          },
+          "l_discount": {"$gte": 0.1, "$lte": 0.3},
+          "l_quantity": {"$lte": 25}
+        }
+      },
+      {
+        "$group": {
+          "_id": 1,
+          "revenue": {"$sum": {"$multiply": ["$l_extendedprice", "$l_discount"]}}
+        }
+      }
+    ]
+
+Q1_2Query: &Q1_2Query
+  Pipeline:
+    [
+      {
+        "$match": {
+          "order.o_orderdate": {
+            "$gte": ISODate("1994-01-01T00:00:00.000Z"),
+            "$lte": ISODate("1994-02-01T00:00:00.000Z")
+          },
+          "l_discount": {"$gte": 0.04, "$lte": 0.06},
+          "l_quantity": {"$gte": 26, "$lt": 33}
+        }
+      },
+      {
+        "$group": {
+          "_id": 1,
+          "revenue": {"$sum": {"$multiply": ["$l_extendedprice", "$l_discount"]}}
+        }
+      }
+    ]
+
+Q1_3Query: &Q1_3Query
+  Pipeline:
+    [
+      {
+        $match: {
+          $and: [
+            { $expr: { $eq: [{$year: '$order.o_orderdate'}, 1994] } },
+            { $expr: { $eq: [{$week: '$order.o_orderdate'}, 6] } },
+            { l_discount: { $gte: 0.04 } },
+            { l_discount: { $lte: 0.06 } },
+            { l_quantity: { $gte: 26 } },
+            { l_quantity: { $lt: 33 } }
+          ]
+        }
+      },
+      {
+        $group: {
+          _id: 1,
+          revenue: { $sum: { $multiply: ['$l_extendedprice', '$l_discount'] } }
+        }
+      }
+    ]
+
+Q2_1Query: &Q2_1Query
+  Pipeline:
+    [
+      { $match: { 'part.p_brand': 'Brand#21' } },
+      {
```

Comment:
The pipeline reorder probably wouldn't buy us much since we only use CSI and no traditional indexes.
As for further denormalizing, I can see why that would improve performance, but I believe we're trying to keep the schema as close to the original SSB benchmark as possible.